hazelcast/serialization/service.py

```diff
@@ -344,7 +344,7 @@ def __init__(
         self._registration_lock = threading.RLock()
         self._int_type_id = _int_type_to_type_id.get(config.default_int_type, None)
 
-        self._compact_types = set(config.compact_serializers.keys())
+        self._compact_types = [c.get_class() for c in config.compact_serializers]
```

Comment:
Does not using a dict make more sense here? We will do a lot of lookups in this data type.

Reply:
It should be a set instead of a list. Will update.

Reply:
It is even better (O(1)) to make it a dict than a set (O(log(n)))

Reply:
Are you sure set lookup is O(log(n))? According to https://wiki.python.org/moin/TimeComplexity it is O(1).

Reply:
Ok I was thinking it is like a set in cpp.